akka-remote/src/main/scala/akka/remote/artery/compress/TopHeavyHitters.scala

```diff
@@ -38,10 +38,9 @@ private[remote] final class TopHeavyHitters[T >: Null](val max: Int)(implicit cl
   /** Returns copy(!) of items which are currently considered to be heavy hitters. */
   def snapshot: Array[T] = {
     val snap = Array.ofDim(max).asInstanceOf[Array[T]]
-    var i: Int = 0
-    val heapIterator = heap.reverseIterator
-    while (heapIterator.hasNext) {
-      val index = heapIterator.next()
+    var i = 0
+    while (i < max) {
+      val index = heap(i)
```

Comment:
> The lowest value are on lower indexes so, taking heap in reverse order would generate array with higher value first.

I suspected that something like this is in play. However, the returned array only has a heap property, it is not really sorted. We don't want to do variable length encoding either, because that would need unaligned accesses in the header and I don't think it is worth it for a few bytes shaven off. Overall, I think it is safe to reverse the order.